nixos/modules/services/network-filesystems/nfsd.nix

```diff
@@ -140,36 +142,37 @@ in
 
     environment.etc.exports.source = exports;
 
-    systemd.services.nfs-server =
-      { enable = true;
-        wantedBy = [ "multi-user.target" ];
+    systemd.services.nfs-server = {
+      wantedBy = [ "multi-user.target" ];
+    };
 
-        preStart =
-          ''
-            mkdir -p /var/lib/nfs/v4recovery
-          '';
-      };
+    systemd.services.nfs-mountd = {
+      restartTriggers = [ exports ];
 
-    systemd.services.nfs-mountd =
-      { enable = true;
-        restartTriggers = [ exports ];
-
-        preStart =
-          ''
-            mkdir -p /var/lib/nfs
-
-            ${optionalString cfg.createMountPoints
-              ''
-                # create export directories:
-                # skip comments, take first col which may either be a quoted
-                # "foo bar" or just foo (-> man export)
-                sed '/^#.*/d;s/^"\([^"]*\)".*/\1/;t;s/[ ].*//' ${exports} \
-                | xargs -d '\n' mkdir -p
-              ''
-            }
-          '';
-      };
+      preStart = optionalString cfg.createMountPoints ''
+        # create export directories:
+        # skip comments, take first col which may either be a quoted
+        # "foo bar" or just foo (-> man export)
+        sed '/^#.*/d;s/^"\([^"]*\)".*/\1/;t;s/[ ].*//' ${exports} \
+        | xargs -d '\n' mkdir -p
+      '';
+    };
 
+    # rpc-statd will drop privileges by changing user from root to the owner of
+    # /var/lib/nfs
+    systemd.tmpfiles.rules = [
+      "d /var/lib/nfs          0700 ${rpcUser} ${rpcUser} - -"
+      "d /var/lib/nfs/recovery 0755 root root             - -"
```

Comment:
Does this throw a silent runtime error with `systemd-tmpfiles` stating unsafe transition? I'm thinking it will... :thinking:

Reply:
Why would it do that?

Reply:
https://github.com/systemd/systemd/issues/13052#issuecomment-511742908